include/rocksdb/io_status.h

```diff
@@ -252,18 +245,8 @@ inline bool IOStatus::operator!=(const IOStatus& rhs) const {
 }
 
 inline IOStatus status_to_io_status(Status&& status) {
-  if (status.ok()) {
-    // Fast path
-    return IOStatus::OK();
-  } else {
-    const char* state = status.getState();
-    if (state) {
-      return IOStatus(status.code(), status.subcode(),
-                      Slice(state, strlen(status.getState()) + 1), Slice());
-    } else {
-      return IOStatus(status.code(), status.subcode());
-    }
-  }
+  IOStatus io_s = *(static_cast<IOStatus*>(&status));
```

Comment:
Can we now have an IOStatus(const Status&) constructor and assignment operator and get rid of this conversion method?

Reply:
I'm concerned about file system code accidentally using Status and returning it. With such a constructor, the compiler will not catch it.